mondey_backend/src/mondey_backend/routers/admin_routers/milestones.py

```diff
@@ -112,11 +156,34 @@ def update_milestone(
         return db_milestone
 
     @router.delete("/milestones/{milestone_id}")
-    def delete_milestone(session: SessionDep, milestone_id: int):
-        milestone = get(session, Milestone, milestone_id)
-        session.delete(milestone)
-        session.commit()
-        return {"ok": True}
+    def delete_milestone(
+        session: SessionDep,
+        milestone_id: int,
+        dry_run: bool = Query(
+            True,
+            description="When true, shows what would be deleted without actually deleting",
+        ),
+    ):
+        affected_answers = count_milestone_answers_for_milestone(session, milestone_id)
+        if not dry_run:
+            milestone = get(session, Milestone, milestone_id)
+            if milestone is None:
+                return 0
```

Comment:
This is not needed as this get utility function already raises a 404 in this case
```suggestion
```